src/seqn/trans.lisp

```diff
@@ -0,0 +1,399 @@
+made into a tuple or a single entry if the codomain is isomorphic
+to (n) for some n"
+  (declare (ignore constraints))
+  (declare (ignore values))
+  (geb.utils:subclass-responsibility obj))
+
+(defun infix-creation (symbol value1 value2)
+  (vamp:make-infix :op symbol
+                   :lhs value1
+                   :rhs value2))
+
+;; Make wire function accessing the wire list
+
+(defmethod to-vampir ((obj id) inputs constraints)
+  "Given a tuple (x1,...,xn) does nothing with it"
+  (declare (ignore constraints))
+  inputs)
+
+(defmethod to-vampir ((obj composition) inputs constraints)
+  "Compile the MCADR after feeding in appropriate
+inputs and then feed them as entries to compiled MCAR"
+  (to-vampir (mcar obj)
+             (to-vampir (mcadr obj) inputs constraints)
+             constraints))
+
+(defmethod to-vampir ((obj parallel-seq) inputs constraints)
+  "Compile MCAR and MCADR and then apppend the tuples"
+  (let* ((mcar (mcar obj))
+         (mcadr (mcadr obj))
+         (lmcar (length (dom mcar))))
+    (append (to-vampir mcar
+                       (subseq inputs 0 lmcar)
+                       constraints)
+            (to-vampir mcadr
+                       (subseq inputs
+                               lmcar)
+                       constraints))))
+
+(defmethod to-vampir ((obj fork-seq) inputs constraints)
+  "Given a tuple (x1,...,xn) copies it twice"
+  (declare (ignore constraints))
+  ;; Since we feed in wires, simply ask for its list of wires and appent
+  (append inputs inputs))
+
+(defmethod to-vampir ((obj drop-nil) inputs constraints)
+  "Drops everything by producing nothing"
+  (declare (ignore inputs constraints))
+  (list (vamp:make-constant :const 0)))
+
+(defmethod to-vampir ((obj remove-right) inputs constraints)
+  "We do not have nul inputs so does nothing"
+  (declare (ignore constraints))
+  (butlast inputs))
+
+(defmethod to-vampir ((obj remove-left) inputs constraints)
+  "We do not have nul inputs so does nothing"
+  (declare (ignore constraints))
+  (cdr inputs))
+
+(defmethod to-vampir ((obj drop-width) inputs constraints)
+  "The compilation does not produce dropping with domain inputs
+wider than codomain ones appropriately. Hence we do not require range
+checks here and simply project"
+  (declare (ignore constraints))
+  inputs)
+
+(defmethod to-vampir ((obj inj-length-left) inputs constraints)
+  "Look at the MCAR. Look at non-null wide entries and place
+0-es in the outputs otherwise ignore"
+  (declare (ignore constraints))
+  (let ((l (length  (mcadr obj))))
+    (if (zerop l)
+        inputs
+        (append inputs (mapcar (lambda (x)
+                          (vamp:make-constant :const x))
+                               (0-list l))))))
+
+(defmethod to-vampir ((obj inj-length-right) inputs constraints)
+  "Look at the MCADR. Look at non-null wide entries and place
+0-es in the outputs "
+  (declare (ignore constraints))
+  (let ((l (length  (mcar obj))))
+    (if (zerop l)
```

Comment:
Can probably cut the length check doing null for the if and using length in the else branch for make-list